Chart changes merged,not deployed,iterate version

georgekaz opened the pull request:
kyverno#2073 - Updates chart version to v1.4.2 - merged 13 days ago but not deployed
kyverno#2081 - Fix Helm release during tag - merged 9 days ago
kyverno#2037 - updated chart contents but not chart version - not released - merged 7 days ago
kyverno#2103 - updated chart contents but not chart version - not released - merged 3 hours ago

Status - main branch shows chart version is v1.4.2, git tag is v1.4.1 and latest helm chart on artifacthub.io/packages/helm/kyverno/kyverno is v1.4.1

This PR just iterates the chart version so that it's built, but I think chart versions should be incremented as part of the normal PR/merge process if chart elements are changed

Signed-off-by: George Kaz <egeorgekaz@gmail.com>

## Related issue

<!--
Please link the GitHub issue this pull request resolves in the format of `#1234`. If you discussed this change
with a maintainer, please mention her/him using the `@` syntax (e.g. `@JimBugwadia`).

If this change neither resolves an existing issue nor has sign-off from one of the maintainers, there is a
chance substantial changes will be requested or that the changes will be rejected.

You can discuss changes with maintainers in the [Kyverno Slack Channel](https://kubernetes.slack.com/).
-->

## What type of PR is this

<!--

> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading white spaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
-->

## Proposed Changes

<!--
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. 

***NOTE***: If this PR results in new or altered behavior which is user facing, you **MUST** read and follow the steps outlined in the [PR documentation guide](pr_documentation.md) and add Proof Manifests as defined below.
-->

### Proof Manifests

<!--
Read and follow the [PR documentation guide](https://github.com/kyverno/kyverno/blob/main/.github/pr_documentation.md) for more details first. This section is for pasting your YAML manifests (Kubernetes resources and Kyverno policies) which allow maintainers to prove the intended functionality is achieved by your PR. Please use proper fenced code block formatting, for example:

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: roles-dictionary
  namespace: default
data:
  allowed-roles: "[\"cluster-admin\", \"cluster-operator\", \"tenant-admin\"]"
```
-->

## Checklist

<!--
Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of
them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
-->

- [] I have read the [contributing guidelines](https://github.com/kyverno/kyverno/blob/main/CONTRIBUTING.md).
- [] I have added tests that prove my fix is effective or that my feature works.
- [] My PR contains new or altered behavior to Kyverno and
  - [] I have added or changed [the documentation](https://github.com/kyverno/website) myself in an existing PR and the link is:
  <!-- Uncomment to link to the PR -->
  <!-- https://github.com/kyverno/website/pull/123 -->
  - [] I have raised an issue in [kyverno/website](https://github.com/kyverno/website) to track the doc update and the link is:
  <!-- Uncomment to link to the issue -->
  <!-- https://github.com/kyverno/website/issues/1 -->
  - [] I have read the [PR documentation guide](https://github.com/kyverno/kyverno/blob/main/.github/pr_documentation.md) and followed the process including adding proof manifests to this PR.

## Further Comments

<!--
If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution
you did and what alternatives you considered, etc...
-->

realshuting commented:
@georgekaz - the current release process is to keep the chart version and release version the same, and the Helm chart is published only when a new release is out.

I agree with any fix for the Helm chart, seems like it's better to publish the new chart with the fix. We can update t[his chart releaser](https://github.com/kyverno/kyverno/pull/2081/files) to be triggered by `charts/kyverno/Chart.yaml` file change.

@JimBugwadia - what do you think?

georgekaz commented:
That would be great. I think the chart has to be releasable in it's own life-cycle separately from the application versions.

JimBugwadia commented:
Sounds good to me. Thanks!

realshuting commented:
Great! @georgekaz - are you interested in submitting a PR for this change? Or else I can create an issue for this.

georgekaz commented:
Sure no problem, I'll see if I can add it to this PR

georgekaz commented:
@realshuting @JimBugwadia how does this look? I've extracted the helm release to it's own file. There was a step just before it which was:

`      
- name: Configure Git
        run: |
          git config user.name "$GITHUB_ACTOR"
          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
`
I can't see the purpose for this so I removed it and didn't copy it to the new helm-release workflow. If you know why it's needed please feel free to feedback.

realshuting commented:
>I can't see the purpose for this so I removed it and didn't copy it to the new helm-release workflow. If you know why it's needed please feel free to feedback.

No, should be ok to remove that. I added it because the other chart releaser didn't work, so just tried different configurations.

realshuting reviewed (approved):
LGTM, thanks @georgekaz!

realshuting commented:
@georgekaz - seems that the releaser failed to publish a new chart https://github.com/kyverno/kyverno/runs/3070078775.

@treydock - have you seen this error in your project before?

treydock commented:
I have not seen that issue before. My last use of that chart releaser was 11 days ago and I use `@master` version so not sure what version I was actually using of the github action: https://github.com/OSC/k8-ldap-configmap/runs/2979170251?check_suite_focus=true.

Not sure if it helps but for my projects I do this when I want to release a new tag:

```
make bump-version tag VERSION=vx.x.x
git push --tags origin main
```

https://github.com/OSC/k8-ldap-configmap/blob/main/Makefile#L64-L72

georgekaz commented:
I can revert to stefanprodan/helm-gh-pages@master like it was master before. I just thought it was better to switch to a tag. I'm not sure how to test this pre-merge so if you have any ideas let me know.

Shall I try this and open a new PR?

georgekaz commented:
Saying that, the error is actually
```
fatal: not a git repository (or any parent up to mount point /github)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
```
so maybe there's some extra config needed that was being done in the other release workflow

treydock commented:
I just noticed, this PR missing this for Helm release workflow:

```
      - name: Checkout
        uses: actions/checkout@v2
        with:
          fetch-depth: 0
```

So the file setup to do Helm release is not actually checking out the code.

https://github.com/kyverno/kyverno/blob/003c865ab9138f8dc899fda983003d35935c3723/.github/workflows/release.yaml#L134-L137

realshuting commented:
> I can revert to stefanprodan/helm-gh-pages@master like it was master before. I just thought it was better to switch to a tag. I'm not sure how to test this pre-merge so if you have any ideas let me know.
> 
> Shall I try this and open a new PR?

Yes, we can merge a fixing PR to test for now.
